distributed/benchmarks/client.py

```diff
@@ -37,3 +33,38 @@ def inc(x):
         total = delayed(sum, pure=True)(L)
 
         total.compute(scheduler=self.client)
+
+
+class WorkerRestrictionsSuite(object):
+
+    def setup(self):
+        cluster = LocalCluster(n_workers=1, threads_per_worker=1,
+                               resources={"resource": 1}, worker_class=Worker)
+        spec = copy.deepcopy(cluster.new_worker_spec())
+        del spec[1]['options']['resources']
```

Comment:
I encourage folks to take a look at using `SpecCluster` directly.  It's not terrible to use manually, and make this sort of thing more explicit.